loki/transformations/array_indexing.py

```diff
@@ -159,6 +159,10 @@ def resolve_vector_notation(routine):
     loop_map = {}
     index_vars = set()
     vmap = {}
+    # find available loops and create map {(lower, upper, step): loop_variable}
+    loops = FindNodes(Loop).visit(routine.body)
+    loop_map = {(loop.bounds.lower, loop.bounds.upper, loop.bounds.step if loop.bounds.step is not None else 1):
+                loop.variable for loop in loops}
```

Comment:
slightly shorter that makes it fit into one line:
```suggestion
    loop_map = {(loop.bounds.lower, loop.bounds.upper, loop.bounds.step or 1): variable for loop in loops}
```